fe/fe-core/src/main/java/org/apache/doris/tablefunction/ParquetMetadataTableValuedFunction.java

```diff
@@ -0,0 +1,474 @@
+    private final String bloomColumn;
+    private final String bloomLiteral;
+
+    public ParquetMetadataTableValuedFunction(Map<String, String> params) throws AnalysisException {
+        Map<String, String> normalizedParams = params.entrySet().stream().collect(Collectors.toMap(
+                entry -> entry.getKey().toLowerCase(),
+                Map.Entry::getValue,
+                (value1, value2) -> value2
+        ));
+        String rawPath = normalizedParams.get(PATH);
+        if (Strings.isNullOrEmpty(rawPath)) {
+            throw new AnalysisException("Property 'path' is required for parquet_meta");
+        }
+        String parsedPath = rawPath.trim();
+        if (parsedPath.isEmpty()) {
+            throw new AnalysisException("Property 'path' must contain at least one location");
+        }
+        List<String> parsedPaths = Collections.singletonList(parsedPath);
+        List<String> normalizedPaths = parsedPaths;
+
+        String rawMode = normalizedParams.getOrDefault(MODE, MODE_METADATA);
+        mode = rawMode.toLowerCase();
+        if (!SUPPORTED_MODES.contains(mode)) {
+            throw new AnalysisException("Unsupported mode '" + rawMode + "' for parquet_meta");
+        }
+        String tmpBloomColumn = null;
+        String tmpBloomLiteral = null;
+        if (MODE_BLOOM_PROBE.equals(mode)) {
+            tmpBloomColumn = normalizedParams.get(COLUMN);
+            if (Strings.isNullOrEmpty(tmpBloomColumn)) {
+                tmpBloomColumn = normalizedParams.get(COLUMN_NAME);
+            }
+            tmpBloomLiteral = normalizedParams.get(VALUE);
+            if (Strings.isNullOrEmpty(tmpBloomColumn) || Strings.isNullOrEmpty(tmpBloomLiteral)) {
+                throw new AnalysisException(
+                        "Missing 'column' or 'value' for mode parquet_bloom_probe");
+            }
+            tmpBloomColumn = tmpBloomColumn.trim();
+            tmpBloomLiteral = tmpBloomLiteral.trim();
+            if (tmpBloomColumn.isEmpty() || tmpBloomLiteral.isEmpty()) {
+                throw new AnalysisException(
+                        "Missing 'column' or 'value' for mode parquet_bloom_probe");
+            }
+        }
+
+        String firstPath = parsedPaths.get(0);
+        String scheme = null;
+        try {
+            scheme = new URI(firstPath).getScheme();
+        } catch (URISyntaxException e) {
+            scheme = null;
+        }
+        Map<String, String> storageParams = new HashMap<>(normalizedParams);
+        // StorageProperties detects provider by "uri", we also hint support by scheme.
+        if (!Strings.isNullOrEmpty(scheme)) {
+            storageParams.put("uri", firstPath);
+        }
+        if (Strings.isNullOrEmpty(scheme) || "file".equalsIgnoreCase(scheme)) {
+            storageParams.put(StorageProperties.FS_LOCAL_SUPPORT, "true");
+        } else {
+            forceStorageSupport(storageParams, scheme.toLowerCase());
```

Comment:
I don't think we need to force set `fs.xx.support`.
`torageProperties.createPrimary` can find the real storage type